### PR TITLE
Add tutorial tracking for going through the welcome portal

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: dd4e06e5debfc83ee4b17bf027949d7c0554a41a available at: https://gist.github.com/96d7c2f41ce639cdf49588cdce1e414e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8950-dd4e06e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8950-dd4e06e.dmg

**Comment by huffman** [comment]
To test:

This requires installing the test-build beta through Steam.

 1. Go through the tutorial
 1. Go back through the tutorial portal and redo the tutorial
 1. Go through the portal to the welcome area
 1. Leave interface
 1. Launch the sandbox
   1. Right click "HIgh Fidelity" in Steam
   1. Click the Local Files tab
   1. Click Browse Local Files
   1. Go to the `server-console` directory
   1. Run `server-console.exe`
 1. Start the current dev build of interface from www.highfidelity.com/dev-download
 1. Go through the tutorial
 1. Go through the welcome portal

Let me know when you are done

File: tutorial/tutorial.js [hunk 1]
```diff
@@ -1027,16 +972,20 @@ TutorialManager = function() {
...
+    var VERSION = 1;
+    var tutorialID;
+
     var self = this;
 
     this.startTutorial = function() {
         currentStepNum = -1;
         currentStep = null;
         startedTutorialAt = Date.now();
+        tutorialID = Script.generateUUID ? Script.generateUUID() : "";
```
Comment (Atlante45) [inline tutorial/tutorial.js]: That looks weird, can you explain what this is doing?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8fe62788311ace5e54b07d5f5326d381bc12fa42 available at: https://gist.github.com/96d7c2f41ce639cdf49588cdce1e414e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8950-8fe6278.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8950-8fe6278.dmg

File: tutorial/tutorial.js (hunk 1 above)
Comment (huffman) [inline tutorial/tutorial.js]: If an old build is running the new content it won't have `Script.generateUUID`. This ensures that it default to an empty string if the function is not available.

Comment (huffman) [inline tutorial/tutorial.js]: I will add a comment to the code

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4148abe28f24822e30d5575da162d247712d8f4e available at: https://gist.github.com/96d7c2f41ce639cdf49588cdce1e414e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8950-4148abe.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8950-4148abe.dmg

**Pull request merged**